### PR TITLE
Day of the week tutorial

### PR DESCRIPTION
**Pull request by gvegayon**
## Implementation

- The day-of-the-week [DOW] effect is implemented as a `RandomVariable` that returns a vector of length 7.
- The `RandomVariable` `latent.HospitalAdmissions` takes care of broadcasting the seven values to match the length of the data (latent infections, in this case).
- The PR replaces how `latent.HospitalAdmissions.sample()` is called; instead of passing a `latent_infections` array, it passes the corresponding `SampledValue` containing the latent infections.
- Since `SampledValues` has `t_start`, we can correct for the offset of the infections wrt the observed data.
- In addition, the constructor of `latent.HospitalAdmissions` receives a new argument indicating what's the first day of the week in the observed data. With the previous point, `latent.HospitalAdmissions.sample()` properly broadcasts the DOW effect.

The tutorial itself recycles the model built during the "Implementing a Hospital Admissions-only Model". 

## Draft Notes (not implemented, but here for future reference)

Current state:

- The day-of-the-week [DOW] effect is applied via `latent.HospitalAdmissions`.
- The sampling method of hospital admissions receives latent infections.
- Those latent infections are a time series equal to the number of observed data points **plus** the infection seeding process.
- Since the seeding process extends the observed time series to the left, the resulting broadcasting should account for that.

```mermaid
flowchart LR
   Seeding[Seeding\nprocess] -->|Input to| LatentI[Latent\nInfections]
   LatentI -->|Input to| HospitalAdmissions
   subgraph HospitalAdmissions["HospitalAdmissions.sample()"]
      dow["day_of_week_effect_rv(\n  n_timepoints=latent_hospital_admissions.size\n)"]
   end
```

To implement:

- We would need to pass on the information about `t_start` resulting from the seeding process.
- Currently, the seeding process has `t_start=self.t_start`, both `None` (see [here](https://github.com/CDCgov/multisignal-epi-inference/blob/cf38ef0d5d904d8f30e12111eaf3403714f6c429/model/src/pyrenew/latent/infections.py#L100-L106)). 
- The latent infections vector should have `t_start=-I0.size`.
- Instead of passing the raw array to `latent.HospitalAdmissions`, we should pass the `SampledValue` object with the `t_start=-I0.size`. 
- With that accessible, instead of passing a `n_timepoints` to the DOW RV, we could pass the `SampledValue` latent infections.
- The DOW RV sample method could then access the `t_start` and the number of time points to sample and adjust accordingly.
- The DOW RV object should know what day of the week the first observation of the sample is; this information could be passed during the construction of the DOW RV.

```mermaid
flowchart LR
   Seeding[Seeding\nprocess] -->|Input to| LatentI[Latent\nInfections]
   LatentI -->|Input to| HospitalAdmissions
   subgraph HospitalAdmissions["HospitalAdmissions.sample()"]
      dow["day_of_week_effect_rv(\n  data=latent_hospital_admissions\n)"]
   end
```

### CODE REVIEW

File: docs/source/tutorials/day_of_the_week.qmd
Comment (damonbayer) [inline docs/source/tutorials/day_of_the_week.qmd:11]: link to the tutorial

Comment (damonbayer) [inline docs/source/tutorials/day_of_the_week.qmd:11]: > In the "Fitting a hospital admissions-only model" tutorial

Link to the tutorial

File: docs/source/tutorials/day_of_the_week.qmd
Comment (damonbayer) [inline docs/source/tutorials/day_of_the_week.qmd:216]: This is a bad idea for identifiability reasons. I think we should use something based on a simplex instead.

File: docs/source/tutorials/day_of_the_week.qmd
Comment (damonbayer) [inline docs/source/tutorials/day_of_the_week.qmd:239]: Would be nice to write this in a generic way that uses the date library to get the day of the week from the start date and compute the correct integer for the offset.

File: docs/source/tutorials/day_of_the_week.qmd
Comment (damonbayer) [inline docs/source/tutorials/day_of_the_week.qmd:195]:
```suggestion
    rng_key=jax.random.key(54),
```
See https://github.com/CDCgov/multisignal-epi-inference/issues/370.

File: docs/source/tutorials/day_of_the_week.qmd
Comment (damonbayer) [inline docs/source/tutorials/day_of_the_week.qmd:279]: > Let's start by reproducing the figure without the day-of-the-week effect, and then plotting the new model with the effect:

Seems to suggest we will be doing more after plotting the figures, but that's all we do.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (damonbayer) [inline model/src/pyrenew/latent/hospitaladmissions.py:24]: I think it makes more sense to just call this `multiplier`. It's not really specific to the day of the week effect.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (damonbayer) [inline model/src/pyrenew/latent/hospitaladmissions.py:199]: Can we pick better names here? https://github.com/CDCgov/multisignal-epi-inference/issues/357

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (damonbayer) [inline model/src/pyrenew/latent/hospitaladmissions.py:235]: Why aren't we using `tile_unitl_n`?

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (damonbayer) [inline model/src/pyrenew/latent/hospitaladmissions.py:231]: Prefer a standard `if-else` ordering.

Comment (damonbayer) [inline model/src/pyrenew/latent/hospitaladmissions.py:231]: Not sure how to think of this… is this saying that we assume `t=0` to be a Monday?

**Review by damonbayer — CHANGES_REQUESTED** [review]
Thanks @gvegayon! Some required changes and some things open for discussion.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (damonbayer) [inline model/src/pyrenew/latent/hospitaladmissions.py:71]: These names could be made more clear.

- `infect_hosp_rate_rv` makes me think of something like a case-fataility-rate, but I think is actually the delay distribution?
- `hosp_report_prob_rv` is not more like an infection-hospitalization-ratio than a reporting probability, I think?

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:231]: Not assuming. It is part of the design. See the documentation.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:235]: This implementation was prior to that function. Will fix.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:231]: This is standard in python, I would argue. What do you think, @dylanhmorris?

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:24]: Sure

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:1]: Not sure I'm following.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:71]: And drop the `_rv` suffix?

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:199]: Even better, got rid of that.

File: docs/source/tutorials/day_of_the_week.qmd
Comment (gvegayon) [reply docs/source/tutorials/day_of_the_week.qmd:11]: Sure!

Comment (gvegayon) [reply docs/source/tutorials/day_of_the_week.qmd:11]: Corrected.

File: docs/source/tutorials/day_of_the_week.qmd
Comment (gvegayon) [reply docs/source/tutorials/day_of_the_week.qmd:216]: Sounds good!

File: docs/source/tutorials/day_of_the_week.qmd
Comment (gvegayon) [reply docs/source/tutorials/day_of_the_week.qmd:264]: Sounds good

File: docs/source/tutorials/day_of_the_week.qmd
Comment (gvegayon) [reply docs/source/tutorials/day_of_the_week.qmd:239]: Fixed

File: docs/source/tutorials/day_of_the_week.qmd
Comment (gvegayon) [reply docs/source/tutorials/day_of_the_week.qmd:195]: done

File: docs/source/tutorials/day_of_the_week.qmd
Comment (gvegayon) [reply docs/source/tutorials/day_of_the_week.qmd:279]: Good point

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:228]: Also, worth noticing `datetime` and `numpy.datetime64` both use Monday as anchor.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (damonbayer) [reply model/src/pyrenew/latent/hospitaladmissions.py:72]: No, sorry. Keep the suffix.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (damonbayer) [reply model/src/pyrenew/latent/hospitaladmissions.py:228]: Can you explain (either in GitHub or a code comment) the logic behind these lines? I find it hard to follow.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (damonbayer) [reply model/src/pyrenew/latent/hospitaladmissions.py:1]: In this model, we assume that a length 7 vector represents a week. So the "fundamental time unit" is one day. Do we do enough to communicate or enforce this? Is that out of scope for this?

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (dylanhmorris) [reply model/src/pyrenew/latent/hospitaladmissions.py:228]: Yes, this is the correct syntax for tuple comprehension with a conditional https://stackoverflow.com/questions/4260280/if-else-in-a-list-comprehension

Comment (damonbayer) [reply model/src/pyrenew/latent/hospitaladmissions.py:228]: But this is not a tuple comprehension? It's just a matter of preference.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:1]: I think is out of scope.

File: model/src/pyrenew/latent/hospitaladmissions.py
Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:228]: It defines the offset in terms of days between 0-7, regardless of the starting date of the data. If the seeding process starts -4 days, `-4 % 7 = 3` that would be Sunday->Saturday->Friday->Thursday. Otherwise, if it is 4 then `4 % 7 = 4`, thus Friday. This would work with bigger numbers too. For instance, adding three weeks, we would have `-25 % 7 = 3`.

Comment (gvegayon) [reply model/src/pyrenew/latent/hospitaladmissions.py:228]: It defines the offset in terms of days between 0-6, regardless of the starting date of the data. If the seeding process starts -4 days, -4 % 7 = 3 that would be Sunday->Saturday->Friday->Thursday. Otherwise, if it is 4 then 4 % 7 = 4, thus Friday. This would work with bigger numbers too. For instance, adding three weeks, we would have -25 % 7 = 3.

File: docs/source/tutorials/day_of_the_week.qmd
Comment (damonbayer) [inline docs/source/tutorials/day_of_the_week.qmd:216]: Why concentration of 1/7? We should probably multiply the Dirichlet by 7, as in the WW model. You can do this with a `TransformedRandomVariable`:

```python
TransformedRandomVariable(
    "hosp_wday_effect",
    DistributionalRV(
        "hosp_wday_effect_raw", dist.Dirichlet(concentration=jnp.ones(7))
    ),
    transforms.AffineTransform(loc=0, scale=7),
)
```

**Comment by gvegayon** [comment]
Ready for review, @damonbayer and @dylanhmorris.

**Review by damonbayer — APPROVED** [review]
Looks good. I think we just want to wait for https://github.com/CDCgov/multisignal-epi-inference/pull/387 so we can avoid having to define `MyDOWEffect`.

We could also merge as is and address that later. Your call @gvegayon.

**Comment by gvegayon** [comment]
> Looks good. I think we just want to wait for #387 so we can avoid having to define `MyDOWEffect`.
> 
> We could also merge as is and address that later. Your call @gvegayon.

I'll wait

**Comment by damonbayer** [comment]
> I'll wait

@gvegayon https://github.com/CDCgov/multisignal-epi-inference/pull/387 is merged

**Comment by gvegayon** [comment]
Just addressed conflicts. Once tests pass, I'll merge.

**Comment by damonbayer** [comment]
@gvegayon The point of waiting for https://github.com/CDCgov/PyRenew/pull/387 was so that your `MyDOWEffect` class could be replaced with a `TransformedRandomVariable`. Please make a new issue for this.

**Comment by gvegayon** [comment]
> @gvegayon The point of waiting for #387 was so that your `MyDOWEffect` class could be replaced with a `TransformedRandomVariable`. Please make a new issue for this. I think it could/should be done in this sprint.

Yes! Sorry I forgot that. Fixing now

**Pull request merged**
